src/DotVVM.Core/Controls/Options/IPagingOptions.cs

```diff
@@ -24,28 +24,29 @@ public interface IPagingOptions
         int TotalItemsCount { get; set; }
 
         /// <summary>
-        /// Determines whether the PageIndex represents the first page.
+        /// Gets whether the <see cref="PageIndex"/> represents the first page.
         /// </summary>
         bool IsFirstPage { get; }
 
         /// <summary>
-        /// Determines whether the PageIndex represents the last page.
+        /// Gets whether the <see cref="PageIndex"/> represents the last page.
         /// </summary>
         bool IsLastPage { get; }
 
         /// <summary>
-        /// Calcualtes the total number of pages.
+        /// Gets the total number of pages.
         /// </summary>
         int PagesCount { get; }
 
         /// <summary>
-        /// Calculates a list of page indexes for the pager controls.
+        /// Gets a list of page indexes near the current page. It can be used to build data pagers.
         /// </summary>
         IList<int> NearPageIndexes { get; }
 
         /// <summary>
-        /// Applies the paging settings to the IQueryable object.
+        /// Applies the paging options to the <paramref name="queryable"/> object.
         /// </summary>
+        /// <param name="queryable">The <see cref="IQueryable{T}" /> to modify.</param>
         IQueryable<T> ApplyToQueryable<T>(IQueryable<T> queryable);
     }
```

Comment:
This interface is either redundant or misleading IMO. Either it will behave like a normal pager with Index + PageSize + TotalCount or it will do some magic in the `ApplyToQueryable` function and the `PageIndex`, `PageSize` will be completely irrelevant. I don't know if we should really remove it, but I don't like it at all.

Reply:
It shouldn't do any magic. But maybe we can move the logic back to `GridViewDataSet`?

Reply:
If it should only sort in the PageIndex/PageCount style, I think that there is no reason have a interface for that...

Reply:
Ok, I don't see any reason to keep it there.